lib/spack/spack/test/config.py

```diff
@@ -167,6 +167,49 @@
         'build_stage:': ['patha', 'pathb']}}
 
 
+packages_merge_low = {
+    'packages': {
+        'foo': {
+            'variants': ['+v1']
+        },
+        'bar': {
+            'variants': ['+v2']
+        }
+    }
+}
+
+packages_merge_high = {
+    'packages': {
+        'foo': {
+            'version': ['a']
+        },
+        'bar': {
+            'version': ['b'],
+            'variants': ['+v3']
+        },
+        'baz': {
+            'version': ['c']
+        }
+    }
+}
+
+
+def test_merge_with_defaults(config, write_config_file):
```

Comment:
@scheibelp Don't know how others feel about it, but for what is worth I started marking regressions like:
```python
@pytest.mark.regression('7924')
def test_merge_with_defaults(config, write_config_file):
   ...
```
This may serve both to document the origin of the test + in case we want to run all the regressions we could do:
```
spack test -m regression
```